src/spec_tools/conform.cljc

```diff
@@ -58,7 +60,17 @@
 (defn string->date [_ x]
   (if (string? x)
     (try
-      #?(:clj  (Date/from (Instant/parse x))
+      #?(:clj (try
+                 ;; Try first parsing the "yyyy-mm-dd" format (ISO Date), because
+                 ;; in this case we have to set the offset to UTC, so the date refers
+                 ;; to the current day, and it's not offsetted by the JVM current timezone
+                 (-> x
+                     (LocalDate/parse DateTimeFormatter/ISO_LOCAL_DATE)
+                     (.atStartOfDay)
+                     (.toInstant ZoneOffset/UTC)
+                     (Date/from))
```

Comment:
I think at some point we should just stop using `java.util.Date` and only use Java 8 Time. Not sure if it should be done now or later.

Reply:
I also thought about getting rid of `Date`, but Clojure's `#inst` implementation [is based on `Date`](https://github.com/clojure/clojure/blob/master/src/clj/clojure/instant.clj), so I'm not so sure anymore if I have an opinion on this.

Reply:
I just realized we probably cannot stop using `java.util.Date`, for the reason I linked above: since `#inst` is `java.util.Date` in clj and `js/Date` in cljs, if we move to Java 8 Time we lose the "print support", so the two implementations don't behave the same anymore (as the cljs would print an `#inst`, but clj would print an `#object[java.time.Instant 0x...`).

Reply:
https://github.com/clojure/clojure/blob/master/src/clj/clojure/core_instant18.clj
https://github.com/clojure/clojure/blob/clojure-1.9.0/src/clj/clojure/core.clj#L6708

`java.time.Instant` satisfies `inst?`

But yeah, seems like there is no print-method. So I guess this can't be changed now.

And I'm kind of surprised that on Cljs side JS Date has to be used instead of Closure Date.

Reply:
Throwing exceptions has a big cost. Just looked at https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/util/StdDateFormat.java while pushing out new version of Jsonista. Would some lookup help to resolve which formatter to use?

Reply:
Yes. Basically the reason for that try/catch is that even though `DatatypeConverter/parseDateTime` can parse the ISO Date format, when it does it includes the timezone information in the resulting `Calendar` object. Since js instead flattens everything to UTC (which I think it's the least surprising behaviour) and since "translating" the timezone of `Calendar` is not trivial (so it might be more expensive than throwing exception), here we are.

This `StdDateFormat/parse` looks promising, I can try this out if we are fine with depending on `jackson-databind`?